apps/aecore/test/aecore_txs_SUITE.erl

```diff
@@ -177,16 +177,23 @@ missing_tx_gossip(Config) ->
     rpc:call(N1, application, set_env, [aecore, ping_interval, 1000000], 5000),
     rpc:call(N2, application, set_env, [aecore, ping_interval, 1000000], 5000),
 
-    %% Ping interval was 500 ms, wait that long
-    timer:sleep(2 * 500),
-
     {ok, TxH1} = add_spend_tx(N1, 1000, 20000 * aec_test_utils:min_gas_price(),  1,  100), %% Ok
     {ok, TxH2} = add_spend_tx(N1, 1000, 20000 * aec_test_utils:min_gas_price(),  2,  100), %% Ok
     {ok, TxH3} = add_spend_tx(N1, 1000, 20000 * aec_test_utils:min_gas_price(),  3,  100), %% Ok
     {ok, TxH4} = add_spend_tx(N1, 1000, 20000 * aec_test_utils:min_gas_price(),  4,  100), %% Ok
     {ok, TxH5} = add_spend_tx(N2, 1000, 20000 * aec_test_utils:min_gas_price(),  5,  100), %% Ok
 
-    {ok, _} = aecore_suite_utils:mine_blocks_until_txs_on_chain(N1, [TxH1, TxH2, TxH3, TxH4], ?MAX_MINED_BLOCKS),
+    none = rpc:call(N2, aec_chain, find_tx_with_location, [TxH1]),
+    none = rpc:call(N2, aec_chain, find_tx_with_location, [TxH4]),
+
+    %% Set the same mining_rate to validate target
+    %% Only needed when chain more than 18 blocks
+    ok = rpc:call(N2, application, set_env, [aecore, expected_mine_rate, aecore_suite_utils:expected_mine_rate()], 5000),
+    {ok, MinedKeyBlocks1} = aecore_suite_utils:mine_blocks_until_txs_on_chain(N1, [TxH1, TxH2, TxH3, TxH4], ?MAX_MINED_BLOCKS),
+    aecore_suite_utils:wait_for_height(N2, aec_blocks:height(lists:last(MinedKeyBlocks1))),
+    ?assertMatch(X when is_binary(X), rpc:call(N2, aec_chain, find_tx_with_location, [TxH1])),
+    ?assertMatch(X when is_binary(X), rpc:call(N2, aec_chain, find_tx_with_location, [TxH4])),
```

Comment:
You are trying to find a location of a transaction encoded by by `aeser_api_encoder:encode/2`. I think this should rather be:

```
    {ok, Tx1Hash} = aeser_api_encoder:safe_decode(tx_hash, TxH1),
    {ok, Tx4Hash} = aeser_api_encoder:safe_decode(tx_hash, TxH4),
    {_BlockHash1, _} = rpc:call(N1, aec_chain, find_tx_with_location, [Tx1Hash]),
    {_BlockHash2, _} = rpc:call(N1, aec_chain, find_tx_with_location, [Tx4Hash]),
```

Reply:
And additionally `assertMatch` on `_BlockHashX` being a binary...

Reply:
Applied in f097666 - thanks.